### PR TITLE
add discrete method for survreg to return Matdist

### PR DESCRIPTION
**Pull request by RaphaelS1**
Tests pass locally, first part of #41 

@jemus42 this will fix speed and size issues you're seeing in parametric

### CODE REVIEW

**Comment by bblodfon** [comment]
Tests should pass now (expect some `lightgbm` ones?). 

There is **still the issue** though when the test set is a bit smaller, eg. 10 observations from the `rat` dataset, there we get slightly different probabilities using `discrete  = TRUE` (new option) vs `FALSE` (old option). Can easily be replicated by adding `predict(task, rows =...)` to the new tests. Maybe not an issue after the conversion? I really don't know. @RaphaelS1 what do you think?

File: tests/testthat/test_survival_surv_parametric.R [hunk 1]
```diff
@@ -144,3 +144,50 @@ test_that("loglogistic", {
   expect_equal(unname(as.matrix(p$distr$cdf(data = t(cdf)))),
     matrix(c(rep(0.2, 50), rep(0.8, 50)), byrow = TRUE, nrow = 2))
 })
+
+task = tsk("rats")
+task$filter(sample(task$nrow, 100))
+
+test_that("discrete - aft", {
+  p_cont = lrn("surv.parametric", discrete = FALSE)$train(task)$predict(task)
```
Comment (bblodfon) [inline tests/testthat/test_survival_surv_parametric.R:152]: if you change here to eg `predict(task, sample(task$row_ids, 10))` you will notice differences

**Comment by RaphaelS1** [comment]
> Tests should pass now (expect some `lightgbm` ones?).
> 
> There is **still the issue** though when the test set is a bit smaller, eg. 10 observations from the `rat` dataset, there we get slightly different probabilities using `discrete = TRUE` (new option) vs `FALSE` (old option). Can easily be replicated by adding `predict(task, rows =...)` to the new tests. Maybe not an issue after the conversion? I really don't know. @RaphaelS1 what do you think?

You can't do ML on 10 rows, so I don't really care about this edge case

**Comment by bblodfon** [comment]
@sebffischer any comments from your side?

**Comment by sebffischer** [comment]
Sorry I am currently sick and will look at it next monday

File: R/learner_survival_surv_parametric.R [hunk 2]
```diff
@@ -137,14 +146,20 @@ LearnerSurvParametric = R6Class("LearnerSurvParametric",
 
     .predict = function(task) {
       pv = self$param_set$get_values(tags = "predict")
-      pred = invoke(.predict_survreg, object = self$model, task = task, learner = self, .args = pv)
+      if (pv$discrete) {
```
Comment (sebffischer) [inline R/learner_survival_surv_parametric.R:149]: this means the discrete parameter should have the required tag

File: R/learner_survival_surv_parametric.R [hunk 3]
```diff
@@ -284,4 +299,57 @@ LearnerSurvParametric = R6Class("LearnerSurvParametric",
   list(lp = as.numeric(lp), distr = distr)
 }
 
+
```
Comment (sebffischer) [inline R/learner_survival_surv_parametric.R:302]: I cannot review this, I guess raphael did this part?

File: tests/testthat/test_paramtest_survival_surv_parametric.R [hunk 4]
```diff
@@ -35,17 +35,18 @@ test_that("paramtest surv.parametric train", {
...
     "task",
     "feature_names",
     "tobit",
-    "learner"
+    "learner",
+    "discrete"
   )
 
-  paramtest = run_paramtest(learner, fun, exclude, tag = "predict")
+  paramtest = run_paramtest(learner, fun = .predict_survreg_continuous, exclude, tag = "predict")
```
Comment (sebffischer) [inline tests/testthat/test_paramtest_survival_surv_parametric.R:49]: fun should be list containing both functions

**Review by sebffischer — CHANGES_REQUESTED** [review]
minor stuff

**Comment by sebffischer** [comment]
please also update the NEWS.md file

File: R/learner_survival_surv_parametric.R (hunk 2 above)
Comment (bblodfon) [reply R/learner_survival_surv_parametric.R:149]: Yes, and it does right?
```
discrete = p_lgl(default = FALSE, tags = "predict")
```

File: R/learner_survival_surv_parametric.R (hunk 3 above)
Comment (bblodfon) [reply R/learner_survival_surv_parametric.R:302]: Yes, I tested it and found some very edge cases that needed fixing and fixed the test code

File: tests/testthat/test_paramtest_survival_surv_parametric.R (hunk 4 above)
Comment (bblodfon) [reply tests/testthat/test_paramtest_survival_surv_parametric.R:49]: refactored

File: R/learner_survival_surv_parametric.R (hunk 2 above)
Comment (sebffischer) [reply R/learner_survival_surv_parametric.R:149]: the "required" tag should be added

File: R/learner_survival_surv_parametric.R (hunk 3 above)
Comment (sebffischer) [reply R/learner_survival_surv_parametric.R:302]: great! :)

File: R/learner_survival_surv_parametric.R (hunk 2 above)
Comment (bblodfon) [reply R/learner_survival_surv_parametric.R:149]: ah, now added

**Comment by RaphaelS1** [comment]
@sebffischer @bblodfon thanks for reviews and help, can we merge?

**Review by bblodfon — APPROVED** [review]
